src/resolve/resolved-dns-scope.c

```diff
@@ -713,6 +713,16 @@ DnsScopeMatch dns_scope_good_domain(
                 if (!dns_scope_get_dns_server(s))
                         return DNS_SCOPE_NO;
 
+                /* Route DS requests to the parent */
+                const char *route_domain = domain;
+                {
+                        DnsResourceKey *t;
+                        DNS_QUESTION_FOREACH(t, question)
+                                if (t->type == DNS_TYPE_DS) {
+                                        dns_name_parent(&route_domain);
```

Comment:
it's ok to ignore errors sometimes, but please make this explicit via casting the result to (void) here. helps static checkers to dtermine if this is an accident or on purpose.